CorsixTH/Lua/graphics.lua

```diff
@@ -208,11 +208,12 @@ end
 
 local function makeGreyscaleGhost(pal)
   local remap = {}
+  local stride = #pal / 256
   -- Convert pal from a string to an array of palette entries
   local entries = {}
-  for i = 1, #pal, 3 do
+  for i = 1, #pal, stride do
     local entry = {pal:byte(i, i + 2)} -- R, G, B at [1], [2], [3]
```

Comment:
Since it now computes stride, shouldn't it here either take `stride` bytes or check that `stride == 3` ?

Reply:
This is the grayscale mapping, I really don't know what to do with alpha here. However, I neglected to include it in palette and I've fixed that thanks to this comment.

Reply:
For clarity here, it checks stride to find if the file it's reading is RGBARGBARGBA, then continues to take [RGB]A[RGB]A; but this spot is not what loads the palette for sprites, it's just looks for similar entries that are gray. I mostly needed it not to crash.